src/psyclone/dynamo0p3.py

```diff
@@ -95,6 +95,10 @@
 # ---------- Fortran datatypes ---------------------------------------------- #
 SUPPORTED_FORTRAN_DATATYPES = ["real", "integer", "logical"]
```

Comment:
It's about here I start getting uneasy. This list of types is used to map from meta-data types to Fortran types, at which point we should be using PSyIR types? However, this would only work properly (as pointed out by @rupertford) if the PSyIR defined primitive types and currently it does not. Not something for this PR though.

Reply:
This list of Fortran types is also used to set and check the mapping from Fortran type to default precision for the generated LFRic PSy-layer code in `configuration.py`. We will be able to modify some of it with the mixed pecision work. In the absence of the PSyIR functionality this was the closest to a consistent approach I could think of, as using the `SUPPORTED_FORTRAN_DATATYPES` list for `MAPPING_DATA_TYPES` acts as a bridge between the kernel descriptors for Fortran data type and the default precisions these types have. It is clunky, I agree, but I also think that it at least provides useful information about the type and precision properties for PSyIR Symbols when we start using them for kernel/algorithm arguments.